File: lang_cpp/PcGaming.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Komputer.cpp"
#include "Performance.cpp"
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "Gpu.cpp"

using namespace std;

class PcGaming : public Komputer, public Performance{

    private:
        vector<string> listDaftarGame;

    
    public:
        PcGaming(){
        }

        PcGaming(string nama, Cpu cpu, vector<Ram> ramList, vector<Harddrive> harddriveList, string kodePerformance, vector<Gpu> gpuList, vector<string> listDaftarGame) : Komputer(nama, cpu, ramList, harddriveList), Performance(kodePerformance, gpuList){
            this->listDaftarGame = listDaftarGame;
        }

        void setListDaftarGame(vector<string> listDaftar){
            this->listDaftarGame = listDaftar;
        }

        vector<string> getListDaftarGame(){
            return listDaftarGame;
        }

        void addGame(string game){
            listDaftarGame.push_back(game);
        }

        ~PcGaming(){
        }

};

File: lang_py/PcGaming.py
from Komputer import Komputer
from Performance import Performance
from Cpu import Cpu
from Gpu import Gpu
from Ram import Ram
from Harddrive import Harddrive

class PcGaming(Komputer, Performance):

    def __init__(self, nama, cpu, ramList, harddriveList, kodePerformance, gpuList, listDaftarGame):
        Komputer.__init__(self, nama, cpu, ramList, harddriveList)
        Performance.__init__(self, kodePerformance, gpuList)
        self.listDaftarGame = listDaftarGame

    def setListDaftarGame(self, listDaftarGame):
        self.listDaftarGame = listDaftarGame

    def getListDaftarGame(self):
        return self.listDaftarGame
    
    def addGame(self, game):
        self.listDaftarGame.append(game)
